plugins/algoltek-aux/fu-algoltek-aux-device.c

```diff
@@ -0,0 +1,560 @@
+/*
+ * Copyright (C) 2024 Algoltek <Algoltek, Inc.>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-algoltek-aux-common.h"
+#include "fu-algoltek-aux-device.h"
+#include "fu-algoltek-aux-firmware.h"
+#include "fu-algoltek-aux-struct.h"
+
+struct _FuAlgoltekAuxDevice {
+	FuDpauxDevice parent_instance;
+	guint8 testoui;
+};
+
+G_DEFINE_TYPE(FuAlgoltekAuxDevice, fu_algoltek_aux_device, FU_TYPE_DPAUX_DEVICE)
+
+FuAlgoltekAuxDevice *
+fu_algoltek_aux_device_new(FuDpauxDevice *device)
+{
+	FuAlgoltekAuxDevice *self = g_object_new(FU_TYPE_ALGOLTEK_AUX_DEVICE, NULL);
+	if (device != NULL)
+		fu_device_incorporate(FU_DEVICE(self), FU_DEVICE(device));
+	return self;
+}
+
+static gboolean
+fu_algoltek_dpaux_device_write(FuAlgoltekAuxDevice *self, GByteArray *buf, guint delayms, GError **error)
+{
+	fu_device_sleep(FU_DEVICE(self), delayms);
+	if (!fu_dpaux_device_write(FU_DPAUX_DEVICE(self),
+				  0x80000,
+				  buf->data,
+				  buf->len,
+				  ALGOLTEK_DEVICE_AUX_TIMEOUT,
+				  error))
+		return FALSE;
+	
+	return TRUE;
+}
+
+static gboolean
+fu_algoltek_dpaux_device_read(FuAlgoltekAuxDevice *self, GByteArray *buf, GError **error)
+{
+	fu_device_sleep(FU_DEVICE(self), 20);
+	if (!fu_dpaux_device_read(FU_DPAUX_DEVICE(self),
+				  0x80010,
+				  buf->data,
+				  buf->len,
+				  ALGOLTEK_DEVICE_AUX_TIMEOUT,
+				  error))
+		return FALSE;
+	
+	return TRUE;
+}
+
+
+static guint16
+gen_crc16(guint16 BData, guint16 crc)
```

Comment:
Can you use fu_crc16_full() instead?

Reply:
Thanks for your advice. I haven't seen any other plugins use these function.
Please wait for me to test it.

Reply:
You're correct, this CRC is different -- BE perhaps. I've renamed it for you.